src/Test/ImplicitMethodsIntegrationTest.php

```diff
@@ -196,35 +285,21 @@ public function testImplicitHeadRequest()
             ->willReturn($finalResponse)
             ->shouldBeCalledTimes(1);
 
-        $routeMiddleware = new PathBasedRoutingMiddleware($router);
-        $handler = new class ($finalHandler->reveal()) implements RequestHandlerInterface
-        {
-            /** @var RequestHandlerInterface */
-            private $handler;
-
-            public function __construct(RequestHandlerInterface $handler)
-            {
-                $this->handler = $handler;
-            }
-
-            public function handle(ServerRequestInterface $request) : ResponseInterface
-            {
-                $middleware = new ImplicitHeadMiddleware(
-                    function () {
-                        return new Response();
-                    },
-                    function () {
-                        return new Stream('php://temp', 'rw');
-                    }
-                );
-
-                return $middleware->process($request, $this->handler);
-            }
-        };
+        $pipeline = new MiddlewarePipe();
+        $pipeline->pipe(new PathBasedRoutingMiddleware($router));
+        $pipeline->pipe($this->getImplicitHeadMiddleware($router));
+        $pipeline->pipe(new MethodNotAllowedMiddleware(function () {
+            return new Response();
+        }));
```

Comment:
Same comments here as above on the `testExplicitRequest()` case: use a complete pipeline, and use response factories that assert failure when invoked for the middleware that should not be generating a response.

Also, I think the `DispatchMiddleware` needs to be here, as a `GET` request will be issued?